source/Coop.IntegrationTests/Clans/ClanTests.cs

```diff
@@ -0,0 +1,292 @@
+﻿using Coop.IntegrationTests.Environment;
+using Coop.IntegrationTests.Environment.Instance;
+using GameInterface.Services.Clans.Messages;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Coop.IntegrationTests.Clans
+{
+    public class ClanTests
+    {
+        internal TestEnvironment TestEnvironment { get; }
+
+        public ClanTests()
+        {
+            // Creates a test environment with 1 server and 2 clients by default
+            TestEnvironment = new TestEnvironment();
+        }
+
+        /// <summary>
+        /// Verify sending ClanNameChange on one client
+        /// Triggers ClanNameChanged on all other clients
+        /// </summary>
+        [Fact]
+        public void ClanNameChange_Publishes_AllClients()
+        {
+            // Arrange
+            var clanId = "TestClan";
+            var clanName = "RealName";
+            var informalName = "FakeName";
+
+            var message = new ClanNameChange(clanId, clanName, informalName);
+
+            var client1 = TestEnvironment.Clients.First();
+
+            var server = TestEnvironment.Server;
+
+            // Act
+            client1.ReceiveMessage(this, message);
+
+            // Assert
+            Assert.Equal(1, server.InternalMessages.GetMessageCount<ClanNameChanged>());
+
+            Assert.Equal(1, client1.InternalMessages.GetMessageCount<ClanNameChanged>());
+
+            foreach (EnvironmentInstance client in TestEnvironment.Clients.Where(c => c != client1))
+            {
+                Assert.Equal(1, client.InternalMessages.GetMessageCount<ClanNameChanged>());
```

Comment:
All of these should be ICommands, we want to check that the state is changed for all clients in a similar manner with these tests. There is another comment that fixes the naming with these